bolt/src/main/java/com/slack/api/bolt/App.java

```diff
@@ -93,7 +107,18 @@
         if (appConfig.isDistributedApp()) {
             middlewareList.add(new MultiTeamsAuthorization(config(), installationService));
         } else if (appConfig.getSingleTeamBotToken() != null) {
-            middlewareList.add(new SingleTeamAuthorization(config(), installationService));
+            try {
+                AuthTestResponse initialAuthTest = client().authTest(r -> r.token(appConfig.getSingleTeamBotToken()));
+                if (initialAuthTest == null || !initialAuthTest.isOk()) {
+                    String error = initialAuthTest != null ? initialAuthTest.getError() : "";
+                    String message = "The token is invalid (auth.test error: " + error + ")";
+                    throw new IllegalArgumentException(message);
```

Comment:
SingleTeamAuthorization should throw an exception for an invalid token when booting. This is consistent behavior with Bolt JS and Python.